packages/backend/src/workers/action.ts

```diff
@@ -124,18 +124,25 @@ worker.on('failed', async (job, err) => {
       job: job.data,
     },
   )
+
+  const flow = await Flow.query()
+    .findById(job.data.flowId)
+    .withGraphFetched('user')
+    .throwIfNotFound()
+
+  let shouldAlwaysSendEmail = false
+  if (flow.config?.errorConfig?.notificationFrequency === 'always') {
+    shouldAlwaysSendEmail = true
+  }
+
   const isEmailSent = await checkErrorEmail(job.data.flowId)
-  if (isEmailSent) {
+  if (!shouldAlwaysSendEmail && isEmailSent) {
```

Comment:
nit: micro-optimization - you can omit the `checkErrorEmail` call by inlining it and using short circuiting.

But others may argue that readability > micro-optimization so up to you.
```
if (!shouldAlwaysSendEmail && await checkErrorEmail(job.data.flowId)) {
```

Reply:
can we also add unit tests for this

Reply:
ok sounds good to me, thanks